### PR TITLE
Haiku fixes

### PR DESCRIPTION
**Pull request by mmuman**
This fixes building on Haiku, and potentially other systems.

### CODE REVIEW

**Comment by mmuman** [comment]
Make sure you update `configure` as I didn't include the changes.

File: configure.ac [hunk 1]
```diff
@@ -305,6 +305,11 @@ AC_CHECK_HEADER(errno.h,
                 HAVE_ERRNO_H=1
                 AC_DEFINE(HAVE_ERRNO_H, 1),
                )
+AC_SUBST(HAVE_SYS_CDEFS_H)
+AC_CHECK_HEADER(sys/cdefs.h,
+                HAVE_SYS_CDEFS_H=1
+                AC_DEFINE(HAVE_SYS_CDEFS_H, 1),
+               )
 AC_SUBST(HAVE_SYS_STAT_H)
```
Comment (mmuman) [inline configure.ac]: Btw, I don't see why you do all this when a simple AC_CHECK_HEADER() does AC_SUBST for you, but maybe that's just legacy.

Comment (mywave82) [reply configure.ac]: If was the first major configure script I made, and I used configure script from other projects as a template. I can do some testing and probably remove some of the AC_SUBST

Comment (mmuman) [reply configure.ac]: Yeah unless you use the HAVE_* directly in configure.ac you should only need `AC_CHECK_HEADER(foo.h)` each time.
If you need to use the result in configure.ac you can do so with `$ac_cv_HAVE_FOO_H` IIRC.

**Comment by mmuman** [comment]
I guess we can close this one now.

**Pull request closed without merging**